src/tqec/block/block.py

```diff
@@ -266,23 +280,30 @@ def instantiate(self) -> cirq.Circuit:
         return circuit
 
     @override
-    def instantiate_without_boundary(self, boundary: BlockDimension) -> cirq.Circuit:
+    def instantiate_without_boundary(self, boundary: Direction3D) -> cirq.Circuit:
         return self.replace_boundary_with_empty_plaquettes(boundary).instantiate()
 
     @override
     def scale_to(self, k: int) -> None:
         self.template.scale_to(k)
 
 
-@dataclass
+_COMPUTATION_DATA_KEY = "tqec_computation_block"
+
+
+# @dataclass
 class Computation:
     """Represents a topological quantum error corrected computation.
 
     The computation is represented by a mapping from a position to the
     computational block whose origin is located at that position.
     """
 
-    blocks: dict[Position3D, ComputationBlock]
+    def __init__(self, blocks: dict[Position3D, ComputationBlock]) -> None:
+        self.blocks = blocks
+        self._graph = nx.Graph()
+        for position, block in blocks.items():
+            self._graph.add_node(position, **{_COMPUTATION_DATA_KEY: block})
```

Comment:
I can see why we will end up needing a graph-like structure, but for the moment you are not linking any node (i.e., the graph is only a collection of nodes without any edge) and not using the graph either, is that right?

Reply:
Yes, this is a relict from my testing a bit.